pkgs/by-name/ne/nextjs-ollama-llm-ui/package.nix

```
@@ -0,0 +1,95 @@
+{ buildNpmPackage
+, fetchFromGitHub
+, google-fonts
+, lib
+  # This is a app can only be used in a browser and starts a web server only accessible at
+  # localhost/127.0.0.1 from the local computer at the given port.
+, defaultHostname ? "127.0.0.1"
+, defaultPort ? 3000
+  # Where to find the Ollama service; this url gets baked into the Nix package
+, ollamaUrl ? "http://127.0.0.1:11434"
+, ...
+}:
+
+let
+  version = "1.0.1";
+in
+buildNpmPackage {
+  pname = "nextjs-ollama-llm-ui";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "jakobhoeg";
+    repo = "nextjs-ollama-llm-ui";
+    rev = "v${version}";
+    hash = "sha256-pZJgiopm0VGwaZxsNcyRawevvzEcK1j5WhngX1Pn6YE=";
+  };
+  npmDepsHash = "sha256-wtHOW0CyEOszgiZwDkF2/cSxbw6WFRLbhDnd2FlY70E=";
+
+  patches = [
+    # Update to a newer nextjs version that buildNpmPackage is able to build.
+    # Remove at nextjs update.
+    ./0001-update-nextjs.patch
+    # nextjs tries to download google fonts from the internet during buildPhase and fails in Nix sandbox.
+    # We patch the code to expect a local font from src/app/Inter.ttf that we load from Nixpkgs in preBuild phase.
+    ./0002-use-local-google-fonts.patch
+    # Modify next.config.js to produce a production "standalone" output at .next/standalone.
+    # This output is easy to package with Nix and run with "node .next/standalone/server.js" later.
+    ./0003-add-standalone-output.patch
+  ];
+
+  # Adjust buildNpmPackage phases with nextjs quirk workarounds.
+  # These are adapted from
+  # https://github.com/NixOS/nixpkgs/blob/485125d667747f971cfcd1a1cfb4b2213a700c79/pkgs/servers/homepage-dashboard/default.nix
+  #######################3
+  preBuild = ''
+    # We have to pass and bake in the Ollama URL into the package
+    echo "NEXT_PUBLIC_OLLAMA_URL=${ollamaUrl}" > .env
+
+    # Replace the google fonts with a local copy from nixpkgs
+    cp "${
+      google-fonts.override { fonts = [ "Inter" ]; }
```

We have a specific package for inter font no need to pull in google fonts for just one font

```suggestion
      inter
```

---

Done